orm/src/main/java/org/hibernate/search/hcore/impl/SearchFactoryReference.java

```diff
@@ -22,16 +20,16 @@
  */
 public final class SearchFactoryReference implements Service {
 
-	private volatile CompletableFuture<ExtendedSearchIntegrator> extendedIntegratorFuture;
+	private volatile ExtendedSearchIntegrator extendedIntegrator;
 
-	public void initialize(CompletableFuture<ExtendedSearchIntegrator> future) {
-		extendedIntegratorFuture = future;
+	public void initialize(ExtendedSearchIntegrator extendedIntegrator) {
+		this.extendedIntegrator = extendedIntegrator;
 	}
 
 	public ExtendedSearchIntegrator getSearchIntegrator() {
-		final CompletableFuture<ExtendedSearchIntegrator> value = this.extendedIntegratorFuture;
+		ExtendedSearchIntegrator value = this.extendedIntegrator;
 		if ( value != null ) {
-			return value.join();
+			return value;
```

Comment:
> we'll need to have faith in the containers?
Having experienced the thrilling task of adding a Wildfly/CDI integration test, I have mixed feelings about this statement :grin:

> I don't expect the application to be considered "deployed" until all frameworks have been initialized
If that's true, we shouldn't bother with blocking calls anywhere, not even in `InitializingIntegratorState` for instance... should we?

Also, you seem to assume that the container itself will take care of integration. As we know, this will probably take years, if it ever happens. So I suspect users will have to take care of that for a time, and they may not have access to as much internals as Wildfly developpers, preventing them to control the execution order as delicately.

Anyway... I must admit I don't clearly see why this commit is useful, I thought it was simply code cleanup. If you think it's necessary, then let's go with it. We'll wait for a ticket confirming there is a problem before we try to fix this problem :)